esphome/components/touchscreen/__init__.py

```diff
@@ -30,6 +40,103 @@
 CONF_SWAP_XY = "swap_xy"
 CONF_TRANSFORM = "transform"
 
+CONF_X_MIN = "x_min"
+CONF_X_MAX = "x_max"
+CONF_Y_MIN = "y_min"
+CONF_Y_MAX = "y_max"
+
+
+def validate_calibration(config):
+    if CONF_CALIBRATION in config:
+        if (
+            cv.int_(config[CONF_CALIBRATION][CONF_X_MIN]) != 0
+            and cv.int_(config[CONF_CALIBRATION][CONF_X_MAX]) != 0
+            and abs(
+                cv.int_(config[CONF_CALIBRATION][CONF_X_MIN])
+                - cv.int_(config[CONF_CALIBRATION][CONF_X_MAX])
+            )
+            < 10
+        ):
+            raise cv.Invalid("Calibration X values difference must be more then 10")
+
+        if (
+            cv.int_(config[CONF_CALIBRATION][CONF_Y_MIN]) != 0
+            and cv.int_(config[CONF_CALIBRATION][CONF_Y_MAX]) != 0
+            and abs(
+                cv.int_(config[CONF_CALIBRATION][CONF_Y_MIN])
+                - cv.int_(config[CONF_CALIBRATION][CONF_Y_MAX])
+            )
+            < 10
+        ):
+            raise cv.Invalid("Calibration Y values difference must be more then 10")
+
+    return config
+
+
+def inherit_transform(touchscreen_config):
+    """
+    If the display configuration has a 'transform' option and the touchscreen doesn't,
+    copy that configuration.
+    If the display configuration has a 'rotation' option and the touchscreen doesn't have a
+    'transform' option, translate the rotation to the needed transform configurations and add
+    it to the touchscreen configuration.
+    """
+    if CONF_TRANSFORM in touchscreen_config:
+        _LOGGER.debug(
+            "'transform' manually specified on the touchscreen. Do not inherit form display"
+        )
+        return touchscreen_config
+
+    fconf = fv.full_config.get()
+    display_id = touchscreen_config[CONF_DISPLAY]
+    display_path = fconf.get_path_for_id(display_id)[:-1]
+    display_config = fconf.get_config_for_path(display_path)
+
+    if CONF_TRANSFORM in display_config:
+        transform = display_config[CONF_TRANSFORM]
```

Comment:
```suggestion
    if transform := display_config[CONF_TRANSFORM]:
```
This is the preferred style for handling optional config.